File: airesources/CSharp/Halite2/hlt/DebugLog.cs
using System.IO;

namespace Halite2.hlt
{
    public class DebugLog
    {
        private TextWriter file;
        private static DebugLog instance;

        private DebugLog(TextWriter f)
        {
            file = f;
        }

        public static void Initialize(TextWriter f)
        {
            instance = new DebugLog(f);
        }

        public static void AddLog(string message)
        {
            try
            {
                instance.file.WriteLine(message);
                instance.file.Flush();
            }
            catch (IOException)
            {
            }
        }
    }
}

File: airesources/Kotlin/src/kotlin/halite/DebugLog.kt
package halite

import java.io.FileWriter
import java.io.IOException

class DebugLog private constructor(private val file: FileWriter) {
    companion object {
        private var instance: DebugLog? = null

        internal fun initialize(f: FileWriter) {
            instance = DebugLog(f)
        }

        fun addLog(message: String) {
            try {
                instance!!.file.write(message)
                instance!!.file.write("\n")
                instance!!.file.flush()
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }
    }
}
